src/shared/components/instant-download/instant-download-fetch/fetchForTranscript.ts

```diff
@@ -137,10 +137,17 @@ export const prepareDownloadParameters = (
     })
     .filter(Boolean) as SingleSequenceFetchParams[];
 
-export const getGenomicSequenceData = (
-  versionedStableId: string,
-  unversionedStableId: string
-) => ({
-  label: `${versionedStableId} genomic`,
-  url: `https://rest.ensembl.org/sequence/id/${unversionedStableId}?content-type=text/plain&type=genomic`
-});
+export const prepareGeneDownloadParameters = (
+  geneData: Awaited<
+    ReturnType<typeof fetchGeneAndTranscriptSequenceMetadata>
+  >['gene']
+) => {
+  const {
+    genomic: { label, start, end, checksum }
+  } = geneData;
+  const url = `/api/refget/sequence/${checksum}?start=${start}&end=${end}&accept=text/plain`;
+  return {
+    label,
+    url
+  };
```

Comment:
We do that same on L119-L123. We could probably have another function responsible for building the url.

Reply:
Extracted the repeated logic into a function called `prepareGenomicDownloadParameters`, and removed the `prepareGeneDownloadParameters` function.